data/qam/mdadm.sh

```diff
@@ -0,0 +1,461 @@
+#!/bin/bash
+
+# check if in /tmp is at least 2048 MB of free disk space
+AVAILABLE_DISK_SPACE=$(df -m /tmp|awk '{print$4}'|grep [[:digit:]])
+if [ "$AVAILABLE_DISK_SPACE" -lt "2048"  ]; then
+    echo "At leat 2G of disk space in /tmp is needed, free disk space or modify script"
+    exit 1
+fi
+
+MD_DEVICE=/dev/md1054
+
+DEV_1=/dev/loop41
+DEV_2=/dev/loop42
+DEV_3=/dev/loop43
+
+IMAGE_SIZE=500M
+RANDOM_DATA_COPY_COUNT=5
+
+LANG=C
+
+TEMP_ROOT=/tmp/mdadm_test
+tempdir=$TEMP_ROOT/$RANDOM
+tempmnt=$tempdir/mnt
+
+# echo command to log and exit on error
+function run
+{
+  echo "# $@"
+  $@ || exit 1
+}
+
+# echo command to log, check for pattern in output and exit on error or pattern not found
+function rungrep
+{
+  pattern=$1
+  shift
+
+  echo "# $@"
+  output=$($@ || exit 1)
+
+  echo "$output"
+  echo "$output" | grep -q "$pattern"
+
+  if [ ! $? = 0 ]
+  then
+    echo "Expected pattern \"$pattern\" not found!"
+    exit 1
+  fi
+}
+
+function passed
+{
+  echo ""
+  echo "==> PASSED"
+  echo ""
+}
+
+function breakdown
+{
+  echo ""
+  echo "Test Breakdown:"
+  echo "---------------"
+  echo ""
+
+  run cd $HOME
+
+  # safety, in case we got interrupted halfway through
+  mount | fgrep $tempmnt && umount $tempmnt
+  if [ -e $MD_DEVICE ] ; then mdadm --stop $MD_DEVICE ; fi
+  #for i in 1 2 3 ; do if [ -e $DEV_$i ] ; then run losetup -d $DEV_$i ; fi ; done
+  for i in 1 2 3 ; do if losetup $(eval echo \$DEV_$i) >/dev/null 2>&1 ; then run losetup -d $(eval echo \$DEV_$i) ; fi ; done
+
+  run rmdir $tempmnt
+  run rm -f $tempdir/* /tmp/mdadm.sh.conf
+  run rmdir $tempdir
+  run rmdir $TEMP_ROOT
+
+  echo ""
+  echo "$result"
+  echo ""
+}
+
+echo ""
+echo "Test Setup:"
+echo "-----------"
+echo ""
+
+run mkdir -p $tempdir
+run mkdir -p $tempmnt
+
+run cd $tempdir
+
+result="FAILED on $(uname -a)"
+
+# only set the trap after the initial creation of the folders is done
+trap breakdown SIGTERM EXIT
+
+run which mdadm
+run mdadm --version
+run rpm -q mdadm
+
+# make sure the md device we want to use is not already used by the system under test
+if [ -e $MD_DEVICE ]
+then
+  echo "Device $MD_DEVICE already in use. Please change the MD_DEVICE variable in this script and try again."
+  exit 1
+fi
+
+for i in 1 2 3
+do
+  if losetup $(eval echo \$DEV_$i) >/dev/null 2>&1
+  then
+    echo "Device $(eval echo \$DEV_$i) already in use. Please change the DEV_$i variable in this script and try again."
+    exit 1
+  fi
+done
+
+echo ""
+echo "Test 1: RAID 0"
+echo "--------------"
+echo ""
+
+result="FAILED Test 1 on $(uname -a)"
+
+for i in 1 2 3
```

Comment:
As we have so many loops with 3 iterations, maybe we can extract it so in case we need 4, we can change single line.
```bash
a='1 2 3'
for i in $a; do echo $i ; done
```

Reply:
I agree, but there will me more changes when we need 4, I would leave it for the future upgrade :sunglasses:

Reply:
Sure ;)